xbmc/utils/LangCodeExpander.cpp

```diff
@@ -208,14 +209,33 @@ bool CLangCodeExpander::ConvertToISO6392T(const std::string& strCharCode, std::s
     {
       if (StringUtils::EqualsNoCase(strCharCode, g_iso639_2[i].name))
       {
-        CodeToString(g_iso639_2[i].code, strISO6392T);
+        CodeToString(g_iso639_2[i].code, strISO6392B);
         return true;
       }
     }
   }
   return false;
 }
 
+bool CLangCodeExpander::ConvertToISO6392T(const std::string& strCharCode, std::string& strISO6392T, bool checkWin32Locales /* = false */)
+{
+
+  if (!ConvertToISO6392B(strCharCode, strISO6392T, checkWin32Locales))
+    return false;
+
+    for (unsigned int index = 0; index < ARRAY_SIZE(LanguageCodes); ++index)
```

Comment:
You could also make LanguageCodes an `std::vector` (or `std::array`) and use a range-based for even more convenience.